middleware/http/bearer/bearer_middleware.go

```diff
@@ -0,0 +1,81 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package bearer
+
+import (
+	"encoding/json"
+
+	"context"
+	"strings"
+
+	"github.com/dapr/components-contrib/middleware"
+	oidc "github.com/coreos/go-oidc"
+	"github.com/valyala/fasthttp"
+)
+
+type bearerMiddlewareMetadata struct {
+	IssuerURL   string `json:"issuerURL"`
+	ClientID	string `json:"clientID"`
+}
+
+// NewBearerMiddleware returns a new oAuth2 middleware
+func NewBearerMiddleware() *Middleware {
+	return &Middleware{}
+}
+
+// Middleware is an oAuth2 authentication middleware
+type Middleware struct {
+}
+
+// GetHandler retruns the HTTP handler provided by the middleware
+func (m *Middleware) GetHandler(metadata middleware.Metadata) (func(h fasthttp.RequestHandler) fasthttp.RequestHandler, error) {
+	meta, err := m.getNativeMetadata(metadata)
+
+	if err != nil {
+		return nil, err
+	}
+
+	provider, err := oidc.NewProvider(context.Background(), meta.IssuerURL)
+	if err != nil {
+		return nil, err
+	}
+
+	verifier := provider.Verifier(&oidc.Config{
+		ClientID: meta.ClientID,
+	})
+
+	return func(h fasthttp.RequestHandler) fasthttp.RequestHandler {
+		return func(ctx *fasthttp.RequestCtx) {
+			authHeader := string(ctx.Request.Header.Peek(fasthttp.HeaderAuthorization))
+			if (!strings.HasPrefix(authHeader, "Bearer ")) {
```

Comment:
Yes. I think so.

Reply:
@mkosieradzki Can you please define `Bearer` as a constant? also is there any chance to be lower case (e.g. `bearer`) it would be safe to check the prefix after converting to lowercase.

Reply:
@youngbupark done